tests/foreman/api/test_syncplan.py

```diff
@@ -493,43 +494,102 @@ def test_positive_repeatedly_add_remove(self):
             self.assertEqual(len(syncplan.read().product), 0)
 
 
-@stubbed()
 class SyncPlanSynchronizeTestCase(APITestCase):
     """Tests specific to synchronizing sync plans."""
 
     @classmethod
     def setUpClass(cls):
-        """Create an organization and products which can be re-used in
-        tests."""
+        """Create an organization which can be re-used in tests."""
         super(SyncPlanSynchronizeTestCase, cls).setUpClass()
         cls.org = entities.Organization().create()
-        cls.products = [
-            entities.Product(organization=cls.org).create() for _ in range(2)
-        ]
 
-    @stubbed()
+    def validate_repo_content(
+            self, repo, content_types, after_sync=True, max_attempts=10):
+        """Check whether corresponding content is present in repository before
+        or after synchronization is performed
+
+        :param repo: Repository entity instance to be validated
+        :param content_types: List of repository content entities that
+            should be validated (e.g. package, erratum, puppet_module)
+        :param bool after_sync: Specify whether you perform validation before
+            synchronization procedure is happened or after
+        :param int max_attempts: That value is basically introduced for slow
+            systems when user sure that sync procedure can take more than 300
+            seconds (30*10 where 10 is a default value for max_attempts
+            variable)
+
+        """
+        for i in range(max_attempts):
+            try:
+                for content in content_types:
+                    if after_sync:
+                        self.assertGreater(repo.content_counts[content], 0)
+                    else:
+                        self.assertFalse(repo.content_counts[content])
+                break
+            except AssertionError:
+                sleep(30)
+                repo = repo.read()
+        else:
+                raise AssertionError(
+                    'Repository contains invalid number of content entities')
+
     @run_only_on('sat')
     @tier2
     def test_positive_synchronize_product(self):
-        """Create a sync plan with one product and sync it.
+        """Create a sync plan with one product and sync it automatically.
 
-        @Assert: A sync plan is created with one product and product can be
-        synchronized.
+        @Assert: Product is synchronized successfully.
 
         @Feature: SyncPlan
         """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
```

Comment:
I don't think that sync procedure connected to product/repo creation procedure. Sync for corresponding product will start only when you assign necessary plan to it and only if current time >= time specified in sync plan, so if it takes more time, so it will start anyway